src/python_testing/TC_AVSM_2_13.py

```diff
@@ -0,0 +1,241 @@
+                "Store this value in aRankedStreamPriorities.",
+            ),
+            TestStep(
+                4,
+                "TH reads RateDistortionTradeOffPoints attribute from CameraAVStreamManagement Cluster on TH_SERVER.",
+                "Store this value in aRateDistortionTradeOffPoints.",
+            ),
+            TestStep(
+                5,
+                "TH reads MinViewport attribute from CameraAVStreamManagement Cluster on TH_SERVER.",
+                "Store this value in aMinViewport.",
+            ),
+            TestStep(
+                6,
+                "TH reads VideoSensorParams attribute from CameraAVStreamManagement Cluster on TH_SERVER.",
+                "Store this value in aVideoSensorParams.",
+            ),
+            TestStep(
+                7,
+                "TH reads MaxEncodedPixelRate attribute from CameraAVStreamManagement Cluster on TH_SERVER.",
+                "Store this value in aMaxEncodedPixelRate.",
+            ),
+            TestStep(
+                8,
+                "TH sets StreamUsage from aRankedStreamPriorities. TH sets VideoCodec, MinResolution, MaxResolution, MinBitRate, MaxBitRate conforming with aRateDistortionTradeOffPoints. TH sets MinFrameRate, MaxFrameRate conforming with aVideoSensorParams. TH sets the MinFragmentLen and MaxFragmentLen = 4000. TH sends the VideoStreamAllocate command with these arguments.",
+                "DUT responds with VideoStreamAllocateResponse command with a valid VideoStreamID.",
+                "Store as `aVideoStreamID`",
+            ),
+            TestStep(
+                9,
+                "TH reads AllocatedVideoStreams attribute from CameraAVStreamManagement Cluster on TH_SERVER",
+                "Verify the number of allocated video streams in the list is 1.",
+            ),
+            TestStep(
+                10,
+                "TH sends the VideoStreamAllocate command with the same arguments as step 8.",
+                "DUT responds with `VideoStreamAllocateResponse` command with a valid `VideoStreamID`",
+                "Verify that this VideoStreamID is the same as `aVideoStreamID`",
+            ),
+            TestStep(
+                11,
+                "TH reads AllocatedVideoStreams attribute from CameraAVStreamManagement Cluster on TH_SERVER",
+                "Verify the number of allocated video streams in the list is 1.",
+            ),
+        ]
+
+    @run_if_endpoint_matches(
+        has_feature(Clusters.CameraAvStreamManagement, Clusters.CameraAvStreamManagement.Bitmaps.Feature.kVideo)
+    )
+    async def test_TC_AVSM_2_13(self):
+        endpoint = self.get_endpoint(default=1)
+        cluster = Clusters.CameraAvStreamManagement
+        attr = Clusters.CameraAvStreamManagement.Attributes
+        commands = Clusters.CameraAvStreamManagement.Commands
+
+        self.step("precondition")
+        # Commission DUT - already done
+
+        self.step(1)
```

Comment:
You don't need to do any of step1 as the decorator already handles that for you.  This will never run if VDO isn't set.

Reply:
Removed the feature check and added a log.